test/rules/use_string_buffer.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N use_string_buffer`
+
+class A {
+  String buffer;
+
+  void foo(int n) {
+    int aux = n;
+    while(aux --> 0) {
+      buffer += 'a'; // LINT
+    }
+  }
+}
```

Comment:
I don't think we can test fields the same way we do local variables because we can't (in the general case) reason about what might happen to them while computing the right-hand side. In the same class `A`, consider adding the following:
```
void bar(int count) {
  while (count-- > 0) {
    buffer += baz();
  }
}
String baz() {
  buffer = buffer + buffer;
  return buffer;
}
```
If we produce a lint in `bar`, that would be a false positive.